js/form_question_selectable.js

```diff
@@ -0,0 +1,273 @@
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+/* global sortable */
+
+class GlpiFormQuestionTypeSelectable {
+
+    /**
+     * The selectable input type.
+     *
+     * @type {string}
+     */
+    #inputType;
+
+    /**
+     * The options container.
+     *
+     * @type {JQuery<HTMLElement>}
+     */
+    #container;
+
+    /**
+     * Create a new GlpiFormQuestionTypeSelectable instance.
+     *
+     * @param {JQuery<HTMLElement>} container
+     */
+    constructor(inputType = null, container = null) {
+        this.#inputType = inputType;
+        this.#container = $(container);
+
+        if (this.#container !== null) {
+            this.#container.children()
+                .each((index, option) => this.#registerOptionListeners($(option)));
+
+            // Register listeners for the empty option
+            this.#container
+                .siblings('div[data-glpi-form-editor-question-extra-details]')
+                .each((index, option) => this.#registerOptionListeners($(option)));
+
+            // Compute the state to update the input names
+            window.glpi_form_editor_controller.computeState();
+
+            // Restore the checked state
+            if (this.#inputType === 'radio') {
+                this.#container
+                    .find('input[type="radio"][checked]')
+                    .prop('checked', true);
+            }
+        }
+    }
+
+    #registerOptionListeners(option) {
+        option
+            .find('input[type="text"]')
+            .on('input', (event) => this.handleOptionChange(event))
+            .on('keydown', (event) => this.handleKeydown(event));
+
+        option
+            .find('i[data-glpi-form-editor-question-option-remove]')
+            .on('click', (event) => this.removeOption(event));
+    }
+
+    enableOptionsSortable() {
+        sortable(this.#container, {
+            // Drag and drop handle selector
+            handle: '[data-glpi-form-editor-question-option-handle]',
+
+            // Accept from others questions
+            acceptFrom: '[data-glpi-form-editor-selectable-question-options]',
+
+            // Placeholder class
+            placeholderClass: 'glpi-form-editor-drag-question-option-placeholder mb-1',
+        });
+    }
+
+    removeOption(event) {
```

Comment:
```suggestion
    #removeOption(event) {
```

I think there are no problems with this method being protected.